static/app/views/detectors/components/forms/metricFormData.tsx

```diff
@@ -0,0 +1,311 @@
+  name: string;
+  projectId: string;
+  query: string;
+  visualize: string;
+}
+
+type MetricDetectorFormFieldName = keyof MetricDetectorFormData;
+
+/**
+ * Enables type-safe form field names.
+ * Helps you find areas setting specific fields.
+ */
+export const METRIC_DETECTOR_FORM_FIELDS = {
+  // Core detector fields
+  aggregate: 'aggregate',
+  query: 'query',
+  kind: 'kind',
+  name: 'name',
+  visualize: 'visualize',
+  environment: 'environment',
+  projectId: 'projectId',
+
+  // Priority level fields
+  initialPriorityLevel: 'initialPriorityLevel',
+  highThreshold: 'highThreshold',
+  mediumThreshold: 'mediumThreshold',
+  resolveThreshold: 'resolveThreshold',
+
+  // Condition fields
+  conditionComparisonAgo: 'conditionComparisonAgo',
+  conditionType: 'conditionType',
+  conditionValue: 'conditionValue',
+
+  // Dynamic fields
+  sensitivity: 'sensitivity',
+  thresholdType: 'thresholdType',
+} satisfies Record<MetricDetectorFormFieldName, MetricDetectorFormFieldName>;
+
+export const DEFAULT_THRESHOLD_METRIC_FORM_DATA = {
+  kind: 'threshold',
+
+  // Priority level fields
+  initialPriorityLevel: PriorityLevel.LOW,
+  conditionType: 'gt',
+  conditionValue: '',
+  conditionComparisonAgo: 60 * 60, // One hour in seconds
+  resolveThreshold: '',
+
+  // Default dynamic fields
+  sensitivity: AlertRuleSensitivity.LOW,
+  thresholdType: AlertRuleThresholdType.ABOVE,
+
+  // Snuba query fields
+  visualize: 'transaction.duration',
+  aggregate: 'count',
+  query: '',
+
+  // Passed in from step 1
+  environment: '',
+  projectId: '',
+  name: '',
+} satisfies MetricDetectorFormData;
+
+/**
+ * Small helper to automatically get the type of the form field.
+ */
+export function useMetricDetectorFormField<T extends MetricDetectorFormFieldName>(
+  name: T
+): MetricDetectorFormData[T] {
+  const value = useFormField(name);
+  return value;
+}
+
+interface NewConditionGroup {
+  conditions: Array<Omit<DataCondition, 'id'>>;
+  logicType: DataConditionGroup['logicType'];
+}
+
+interface NewDataSource {
+  aggregate: string;
+  dataset: string;
+  environment: string | null;
+  eventTypes: string[];
+  query: string;
+  queryType: number;
+  timeWindow: number;
+}
+
+export interface NewMetricDetector {
+  conditionGroup: NewConditionGroup;
+  // TODO: config types don't exist yet
+  config: {
+    // TODO: what is this
+    detection_type: any;
+    threshold_period: number;
+  };
+  dataSource: NewDataSource; // Single data source object (not array)
+  detectorType: Detector['type'];
+  name: string;
+  projectId: Detector['projectId'];
+}
+
+// Map PriorityLevel to DetectorPriorityLevel integer values
+// Note: Metric detectors only support MEDIUM and HIGH priority levels
+const PRIORITY_LEVEL_TO_DETECTOR_PRIORITY = {
+  [PriorityLevel.LOW]: 25, // DetectorPriorityLevel.LOW
+  [PriorityLevel.MEDIUM]: 50, // DetectorPriorityLevel.MEDIUM
+  [PriorityLevel.HIGH]: 75, // DetectorPriorityLevel.HIGH
+} as const;
```

Comment:
Could we just use `DetectorPriorityLevel.LOW` instead of `25` here so we don't repeat it?